sdk/clientcore/core/src/main/java/io/clientcore/core/http/pipeline/OAuthBearerTokenAuthenticationPolicy.java

```diff
@@ -67,8 +67,8 @@ public void authorizeRequest(HttpRequest httpRequest, OAuthTokenRequestContext c
     @Override
     public Response<BinaryData> process(HttpRequest httpRequest, HttpPipelineNextPolicy next) {
         if (!"https".equals(httpRequest.getUri().getScheme())) {
-            throw LOGGER.logThrowableAsError(
-                new RuntimeException("Token credentials require a URL using the HTTPS protocol scheme"));
+            throw LOGGER.throwableAtError()
+                .log("Token credentials require a URL using the HTTPS protocol scheme.", IllegalArgumentException::new);
```

Comment:
Throwing this somewhere random

Should we have two `log` methods for this?

`log(String, Function<String, T>>` and `log(String, Throwable, BiFunction<String, Throwable, T>)`?

That way we don't need to ignore `Throwable` in exception types that don't support causal exceptions.

Reply:
great idea, done